### PR TITLE
Remove explicit heights on form elements

### LINKED ISSUE #18842: Temporal inputs are 2px taller than other textual inputs in Chrome
In Chrome, `<input class="form-control">`s with a `type` of `datetime-local`, `date`, `month`, `week`, or `time` are 40px tall, but `<input type="text" class="form-control">` (and similar) are 38px tall. They ought to all match.
Looks like the temporal inputs use Flexbox internally; that's likely a factor.

### PR DESCRIPTION
**Pull request by MartijnCuppens**
Input heights were initially added in https://github.com/twbs/bootstrap/pull/26820 to fix https://github.com/twbs/bootstrap/issues/18842. Select heights were added to fix https://bugzilla.mozilla.org/show_bug.cgi?id=1560824.

But:
- The additional height in https://github.com/twbs/bootstrap/pull/26820 was caused by pseudo elements, something I fixed in https://github.com/twbs/bootstrap/pull/30269.
- In the meantime the select issue in FF is changed when you explicitly set the line height (https://bugzilla.mozilla.org/show_bug.cgi?id=1560824#c14), making setting the height redundant

However, removing the explicit height would reintroduce https://github.com/twbs/bootstrap/issues/23307. Luckily, date inputs have a `::-webkit-datetime-edit` of which we can set the height in `em` 🎉

This means, we can safely remove those (min-)heights that seem to cause some issues like https://github.com/twbs/bootstrap/pull/26820#issuecomment-565485267 or https://github.com/twbs/bootstrap/pull/31955#issuecomment-716101054 (this fixes issues when zoomed in/out)

I've kept a min-height for textareas, to keep support for https://github.com/twbs/bootstrap/pull/29124

I didn't apply this change to `.form-file`, since this will introduce conflicts with https://github.com/twbs/bootstrap/pull/31955, therefor I'll keep this in draft.

### CODE REVIEW

**Review by mdo — APPROVED** [review]
Hell. Yes. :shipit:

**Comment by MartijnCuppens** [comment]
Don't merge this yet, I'm doing some tests withs input field heights to confirm this works (already found a bug in the current implementation which was solved in this PR, lol)

**Comment by MartijnCuppens** [comment]
Ok, I made a codepen to check browsers with and it looks like we're ok with this change: https://codepen.io/MartijnCuppens/pen/JjKpbGd?editors=1010

I actually even found out there's a bug with the current implementation: https://codepen.io/MartijnCuppens/pen/PozQbqJ?editors=1010

File: scss/forms/_form-control.scss [hunk 1]
```diff
@@ -45,6 +44,13 @@
     }
   }
 
+  // Add some height to date inputs on IOS
+  // https://github.com/twbs/bootstrap/issues/23307
+  &::-webkit-date-and-time-value {
```
Comment (XhmikosR) [inline scss/forms/_form-control.scss:49]: I suppose this in not in the specs, right?

Comment (MartijnCuppens) [reply scss/forms/_form-control.scss:49]: Nope. This is a bug in Safari's default CSS (see https://opensource.apple.com/source/WebCore/WebCore-7602.4.6/css/html.css.auto.html, input::-webkit-inner-spin-button)

This bug has been reported here and I've just suggested the fix on that issue: https://bugs.webkit.org/show_bug.cgi?id=198959

Comment (XhmikosR) [reply scss/forms/_form-control.scss:49]: Maybe we should add a TODO comment then?

Comment (MartijnCuppens) [reply scss/forms/_form-control.scss:50]: Done

**Comment by patrickhlauke** [comment]
i *think* this change now made `<input type="color">` revert to its odd/small size - see for instance https://deploy-preview-32020--twbs-bootstrap.netlify.app/docs/5.0/forms/form-control/#color (from my other PR)

**Pull request merged**